packages/superset-ui-control-utils/package.json

```diff
@@ -28,5 +28,10 @@
   "peerDependencies": {
     "@superset-ui/translation": "^0.13",
     "@superset-ui/validator": "^0.13"
+  },
+  "dependencies": {
+    "lodash": "^4.17.15",
+    "react": "^16.13.1",
+    "react-bootstrap": "^1.0.1"
```

Comment:
Re: dependencies

1. These should be `peerDependencies`, as they are shared by multiple `superset-ui` packages and should be upgraded all at once.
2. We are not using `react-bootstrap@1` yet. It's for Bootstrap v4, but we are still stuck with Boostrap v3. Check `incubator-superset` for property versions for `react-bootstrap` and `@types/react-bootstrap`.
3. Can we get rid of `lodash`? Is `kebabCase` absolutely necessary? Could you investigate why do we needed an ID for the tooltip in the first place? It's the best if we could just remove it. Less dependencies the better.

Reply:
Regarding 3, I think that kind of effort is better left until after we've moved things from superset. This should be as clean a refactor as possible, just moving code into superset-ui.

Reply:
`react` in particular should always be `peerDependencies` in any library.

Reply:
I also don't agree with making these peer dependencies, because the code directly depends on them. Peer dependencies aren't meant to be used for optimization, npm does that automatically when version numbers of dependencies are compatible with each other.

Reply:
I agree with making them `dependencies` as-is except `react`

Reply:
True, npm might be able to remove duplicates, but I think there is also the implication that what if it is not compatible? Are we really OK with having multiple versions of the dependent libraries in our app?

These `superset-ui` packages cannot work outside of Superset or Superset UI demos, so they should not introduce their own large non-optional dependencies. Using `peerDependencies` kind of forces package maintainers to keep the dependencies up to date.

Reply:
> True, npm might be able to remove duplicates, but I think there is also the implication of what if it is not compatible? Are we really OK with having multiple versions of the dependent libraries in our app?

I think that is how all packages work tho. I am ok having multiple versions of the dependent instead of worrying about breaking anything if bump version of one dependencies.

> These superset-ui packages cannot work outside of Superset or Superset UI demos, so they should not introduce their own large non-optional dependencies. Using peerDependencies kind of forces package maintainers to keep the dependencies up to date.

These packages can work outside of Superset and Superset UI demos. They are independent packages on their own (number format, time format, color, etc.) which do not need Superset to function. I have them as dependencies in other projects and have always think of them independent from `incubator-superset`. They should have their own dependencies.

Reply:
All other superset-ui packages have things listed under `dependencies`, including lodash. Is this different somehow?

Reply:
I am proposing we proceed with moving `react` to `peerDependencies` and leave the remaining in `dependencies` unless there is objection?

Reply:
No objection here. I actually already moved react, but haven't pushed it yet.